Problem Uploading Sample (missing storage_provider validation)

**Environment information**
* MWDB version (from `/about`): **v2.7.0**
* Installation method:
  - [ ] mwdb.cert.pl service
  - [x] From PyPi (`pip install mwdb-core`)
  - [ ] From docker-compose
  - [ ] Other (please explain)
* Plugins installed: **-**

**Behaviour the bug (what happened?)**

Trying to upload sample, getting error "**Internal Server Error**". Logs as below:
```
(venv) fossil@mwfossil-box:~/mwdb$ mwdb-core run -h 192.168.42.130 -p 5000
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on http://192.168.42.130:5000/ (Press CTRL+C to quit)
[INFO] Thread-3 - app.log_request:141 - request - path:/upload - arguments:{} - method:GET - status:304 - response_time:0:00:00.000596 - response_size:None
192.168.42.1 - - [18/Jun/2022 19:05:22] "GET /upload HTTP/1.1" 304 -
[INFO] Thread-2 - app.log_request:141 - request - path:/static/css/4.afb3e8c4.chunk.css - arguments:{} - method:GET - status:304 - response_time:0:00:00.000352 - response_size:None
192.168.42.1 - - [18/Jun/2022 19:05:22] "GET /static/css/4.afb3e8c4.chunk.css HTTP/1.1" 304 -
[INFO] Thread-4 - app.log_request:141 - request - path:/static/css/main.12f1f60b.chunk.css - arguments:{} - method:GET - status:304 - response_time:0:00:00.000538 - response_size:None
192.168.42.1 - - [18/Jun/2022 19:05:22] "GET /static/css/main.12f1f60b.chunk.css HTTP/1.1" 304 -
[INFO] Thread-5 - app.log_request:141 - request - path:/static/js/4.7fc2e803.chunk.js - arguments:{} - method:GET - status:304 - response_time:0:00:00.000319 - response_size:None
192.168.42.1 - - [18/Jun/2022 19:05:22] "GET /static/js/4.7fc2e803.chunk.js HTTP/1.1" 304 -
[INFO] Thread-6 - app.log_request:141 - request - path:/static/js/main.b6a58194.chunk.js - arguments:{} - method:GET - status:304 - response_time:0:00:00.000506 - response_size:None
192.168.42.1 - - [18/Jun/2022 19:05:22] "GET /static/js/main.b6a58194.chunk.js HTTP/1.1" 304 -
[INFO] Thread-7 - app.log_request:141 - request - path:/static/media/logo.787cfdb1.png - arguments:{} - method:GET - status:304 - response_time:0:00:00.000348 - response_size:None
192.168.42.1 - - [18/Jun/2022 19:05:22] "GET /static/media/logo.787cfdb1.png HTTP/1.1" 304 -
[INFO] Thread-9 - app.log_request:141 - request - path:/api/remote - arguments:{} - method:GET - status:200 - response_time:0:00:00.058049 - response_size:16
[INFO] Thread-8 - app.log_request:141 - request - path:/api/server - arguments:{} - method:GET - status:200 - response_time:0:00:00.065127 - response_size:304
192.168.42.1 - - [18/Jun/2022 19:05:22] "GET /api/server HTTP/1.1" 200 -
192.168.42.1 - - [18/Jun/2022 19:05:22] "GET /api/remote HTTP/1.1" 200 -
[INFO] Thread-10 - auth.post:420 - Session token refreshed - user:admin
[INFO] Thread-10 - app.log_request:141 - request - path:/api/auth/refresh - arguments:{} - method:POST - status:200 - response_time:0:00:00.050503 - response_size:922
192.168.42.1 - - [18/Jun/2022 19:05:22] "POST /api/auth/refresh HTTP/1.1" 200 -
[INFO] Thread-11 - app.log_request:141 - request - path:/api/share - arguments:{} - method:GET - status:200 - response_time:0:00:00.009166 - response_size:60
192.168.42.1 - - [18/Jun/2022 19:05:22] "GET /api/share HTTP/1.1" 200 -
[INFO] Thread-12 - app.log_request:141 - request - path:/api/attribute - arguments:{'access': 'set'} - method:GET - status:200 - response_time:0:00:00.021539 - response_size:30
192.168.42.1 - - [18/Jun/2022 19:05:22] "GET /api/attribute?access=set HTTP/1.1" 200 -
[INFO] Thread-13 - app.log_request:141 - request - path:/favicon.ico - arguments:{} - method:GET - status:304 - response_time:0:00:00.000398 - response_size:None
192.168.42.1 - - [18/Jun/2022 19:05:22] "GET /favicon.ico HTTP/1.1" 304 -
[ERROR] Thread-14 - service.error_router:76 - Unhandled exception occurred
Traceback (most recent call last):
  File "/home/fossil/mwdb/venv/lib/python3.8/site-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/fossil/mwdb/venv/lib/python3.8/site-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/home/fossil/mwdb/venv/lib/python3.8/site-packages/flask_restful/__init__.py", line 467, in wrapper
    resp = resource(*args, **kwargs)
  File "/home/fossil/mwdb/venv/lib/python3.8/site-packages/flask_limiter/extension.py", line 1036, in __inner
    return current_app.ensure_sync(obj)(*a, **k)
  File "/home/fossil/mwdb/venv/lib/python3.8/site-packages/flask_limiter/extension.py", line 1036, in __inner
    return current_app.ensure_sync(obj)(*a, **k)
  File "/home/fossil/mwdb/venv/lib/python3.8/site-packages/flask_limiter/extension.py", line 1036, in __inner
    return current_app.ensure_sync(obj)(*a, **k)
  [Previous line repeated 5 more times]
  File "/home/fossil/mwdb/venv/lib/python3.8/site-packages/flask/views.py", line 84, in view
    return current_app.ensure_sync(self.dispatch_request)(*args, **kwargs)
  File "/home/fossil/mwdb/venv/lib/python3.8/site-packages/flask_restful/__init__.py", line 582, in dispatch_request
    resp = meth(*args, **kwargs)
  File "/home/fossil/mwdb/venv/lib/python3.8/site-packages/mwdb/resources/__init__.py", line 47, in endpoint
    return f(*args, **kwargs)
  File "/home/fossil/mwdb/venv/lib/python3.8/site-packages/mwdb/resources/__init__.py", line 31, in endpoint
    return f(*args, **kwargs)
  File "/home/fossil/mwdb/venv/lib/python3.8/site-packages/mwdb/resources/file.py", line 187, in post
    return self.create_object(obj["options"])
  File "/home/fossil/mwdb/venv/lib/python3.8/site-packages/mwdb/resources/object.py", line 125, in create_object
    item, is_new = self._create_object(
  File "/home/fossil/mwdb/venv/lib/python3.8/site-packages/mwdb/resources/file.py", line 34, in _create_object
    return File.get_or_create(
  File "/home/fossil/mwdb/venv/lib/python3.8/site-packages/mwdb/model/file.py", line 147, in get_or_create
    with open(file_obj._calculate_path(), "wb") as f:
FileNotFoundError: [Errno 2] No such file or directory: 'e/d/1/4/ed14db6907dfa9942967ef37a60f09328078322aa14d835d5001fb5796a39359'
[INFO] Thread-14 - app.log_request:141 - request - path:/api/file - arguments:{} - method:POST - status:500 - response_time:0:00:00.031207 - response_size:37
192.168.42.1 - - [18/Jun/2022 19:05:26] "POST /api/file HTTP/1.1" 500 -
```

**Expected behaviour**

To successfully upload sample without error.

**Screenshots**

![image](https://user-images.githubusercontent.com/3533227/174453036-b9188149-6ede-4543-83f6-33fe8b3b9c38.png)

--- a/mwdb/core/config.py
+++ b/mwdb/core/config.py
@@ -1,6 +1,6 @@
 import os
 from enum import Enum
-from typing import List, Optional
+from typing import List
 
 from typedconfig import Config, group_key, key, section
 from typedconfig.source import EnvironmentConfigSource, IniFileConfigSource
@@ -18,7 +18,7 @@ def list_of_str(v: str) -> List[str]:
 
 def path(v: str) -> str:
     if not v:
-        raise ValueError(f"Path can't be empty")
+        raise ValueError("Path can't be empty")
     v = os.path.abspath(v)
     if not os.path.exists(v):
         raise ValueError(f"Path {v} doesn't exist")

--- a/mwdb/core/config.py
+++ b/mwdb/core/config.py
@@ -8,15 +8,15 @@ from typedconfig.source import EnvironmentConfigSource, IniFileConfigSource
 from mwdb.paths import mail_templates_dir
 
 
-def intbool(v) -> bool:
+def intbool(v: str) -> bool:
     return bool(int(v))
 
 
-def list_of_str(v) -> List[str]:
+def list_of_str(v: str) -> List[str]:
     return [el.strip() for el in v.split(",") if el.strip()]
 
 
-def path(v) -> Optional[str]:
+def path(v: str) -> str:
     if not v:
         raise ValueError(f"Path can't be empty")
     v = os.path.abspath(v)

--- a/mwdb/core/config.py
+++ b/mwdb/core/config.py
@@ -18,7 +18,7 @@ def list_of_str(v) -> List[str]:
 
 def path(v) -> Optional[str]:
     if not v:
-        return None
+        raise ValueError(f"Path can't be empty")
     v = os.path.abspath(v)
     if not os.path.exists(v):
         raise ValueError(f"Path {v} doesn't exist")
@@ -30,9 +30,9 @@ class StorageProviderType(Enum):
     S3 = "S3"
 
 
-def storage_provider_from_str(v: str) -> Optional[StorageProviderType]:
+def storage_provider_from_str(v: str) -> StorageProviderType:
     if not v:
-        return None
+        return StorageProviderType.DISK
 
     v = v.upper()
     try:
@@ -66,7 +66,7 @@ class MWDBConfig(Config):
     request_timeout = key(cast=int, required=False, default=20000)
     # Which storage provider to use (options: disk or s3)
     storage_provider = key(
-        cast=storage_provider_from_str, required=False, default="disk"
+        cast=storage_provider_from_str, required=False, default=StorageProviderType.DISK
     )
     # File upload timeout
     file_upload_timeout = key(cast=int, required=False, default=60000)
@@ -97,21 +97,20 @@ class MWDBConfig(Config):
 
     enable_plugins = key(cast=intbool, required=False, default=True)
     # List of plugin names to be loaded, separated by commas
-    plugins = key(cast=list_of_str, required=False, default="")
+    plugins = key(cast=list_of_str, required=False, default=[])
     # Directory that will be added to sys.path for plugin imports
     # Allows to load local plugins without installing them in site-packages
     local_plugins_folder = key(cast=path, required=False, default=None)
     # Auto-discover plugins contained in local_plugins_folder
     local_plugins_autodiscover = key(cast=intbool, required=False, default=False)
 
-    remotes = key(cast=list_of_str, required=False, default="")
+    remotes = key(cast=list_of_str, required=False, default=[])
 
     enable_rate_limit = key(cast=intbool, required=False, default=False)
     enable_registration = key(cast=intbool, required=False, default=False)
     enable_maintenance = key(cast=intbool, required=False, default=False)
     enable_hooks = key(cast=intbool, required=False, default=True)
     enable_karton = key(cast=intbool, required=False, default=False)
-    # Feature flag: OIDC is under development
     enable_oidc = key(cast=intbool, required=False, default=False)
 
     mail_smtp = key(cast=str, required=False)

--- a/requirements.txt
+++ b/requirements.txt
@@ -20,7 +20,7 @@ click-default-group==1.2.2
 PyYAML==5.4
 redis==3.2.1
 minio==7.1.0
-typed-config==0.2.1
+typed-config==1.1.0
 karton-core==4.4.1
 Authlib==0.15.4
 prance==0.21.8.0  # apispec unpinned dependency

